simple_certmanager/admin.py

```diff
@@ -1,19 +1,108 @@
+from io import BytesIO
+from zipfile import ZipFile
+
 from django.contrib import admin
+from django.http import HttpResponse, HttpResponseRedirect
+from django.urls import reverse
 from django.utils.translation import gettext_lazy as _
 
 from privates.admin import PrivateMediaMixin
 
-from .forms import CertificateAdminForm
-from .models import Certificate
+from .forms import CertificateAdminForm, SigningRequestAdminForm
+from .models import Certificate, SigningRequest
 from .utils import suppress_cryptography_errors
 
 
+def download_csr(modeladmin, request, queryset):
+    if queryset.count() > 1:
+        zip_file = BytesIO()
+        with ZipFile(zip_file, "w") as zipf:
+            for csr in queryset:
+                csr_content = csr.csr
+                csr_filename = f"{csr.common_name}_csr.pem"
+                zipf.writestr(csr_filename, csr_content)
+        zip_file.seek(0)
+        response = HttpResponse(zip_file, content_type="application/zip")
+        response["Content-Disposition"] = 'attachment; filename="csr.zip"'
+        return response
+    elif queryset.count() == 1:
+        csr = queryset.first().csr
+        response = HttpResponse(csr, content_type="application/x-pem-file")
+        response["Content-Disposition"] = (
+            f'attachment; filename="{queryset.first().common_name}_csr.pem"'
+        )
+        return response
+    else:
+        modeladmin.message_user(
+            request,
+            "No CSR selected.",
+        )
+
+
+@admin.register(SigningRequest)
+class SigningRequestAdmin(PrivateMediaMixin, admin.ModelAdmin):
+    model: type[SigningRequest]
+    form = SigningRequestAdminForm
+    fields = (
+        "common_name",
+        "organization_name",
+        "country_name",
+        "state_or_province_name",
+        "email_address",
+        "private_key",
+        "csr",
+        "certificate",
+    )
+    list_display = (
+        "common_name",
+        "organization_name",
+        "country_name",
+        "state_or_province_name",
+        "email_address",
+    )
+    list_filter = ("organization_name", "state_or_province_name")
+    readonly_fields = ("csr",)
+    search_fields = ("common_name", "organization_name")
+    private_media_fields = ("private_key",)
+    private_media_no_download_fields = ("private_key",)
+    actions = [download_csr]
```

Comment:
neat

I'd also expect a link on the detail page and in the admin changelist page for each record